### ISSUE TITLE
tslint@next does not recognize some reassignation

### ISSUE THREAD

**Issue by SaschaNaz** [palantir/tslint#1898]
### Bug Report

- __TSLint version__: `tslint@next`
- __TypeScript version__: `typescript@next`
- __Running TSLint via__: Node.js API

#### TypeScript code being linted

From https://github.com/Microsoft/TypeScript/pull/13006:

https://github.com/SaschaNaz/TypeScript/blob/27a60e4580076b544fa7f822df84414b7ea2c357/src/services/navigationBar.ts#L32

```ts
export function getNavigationTree(sourceFile: SourceFile): NavigationTree {
    curSourceFile = sourceFile;
    const result = convertToTree(rootNavigationBarNode(sourceFile));
    curSourceFile = undefined;
    return result;
}
let curSourceFile: SourceFile;
// Identifier 'curSourceFile' is never reassigned; use 'const' instead of 'let'.
```

https://github.com/SaschaNaz/TypeScript/blob/27a60e4580076b544fa7f822df84414b7ea2c357/src/harness/unittests/session.ts#L395

```ts
let response: protocol.Response;
try {
    ({ response } = this.executeCommand(msg));
}
//  Identifier 'response' is never reassigned; use 'const' instead of 'let'.
```

with `tslint.json` configuration: https://github.com/Microsoft/TypeScript/blob/b15d1449550759f57759097a802abf14943e69cf/tslint.json

#### Actual behavior

Identifier '(variable name)' is never reassigned; use 'const' instead of 'let'.

#### Expected behavior

The variable is being reassigned so tslint shouldn't error out

**Comment by nchen63** [comment]
fixed by #1906 and #1908
